src/libraries/System.Private.CoreLib/src/System/Reflection/Assembly.cs

```diff
@@ -218,12 +218,17 @@ public override string ToString()
         }
 
         // internal test hook
-        private static bool s_forceNullEntryPoint;
+        private static Assembly? s_overriddenEntryAssembly;
```

Comment:
You will need to replace `s_forceNullEntryPoint` uses in the tests with calls of this API.

https://github.com/search?q=repo%3Adotnet%2Fruntime%20s_forceNullEntryPoint&type=code